src/widget/modal.svelte

```diff
@@ -0,0 +1,277 @@
+<script context="module" lang="ts">
+  import {
+    Config,
+    FRUser,
+    type GetTokensOptions,
+    SessionManager,
+    TokenManager,
+    UserManager,
+    HttpClient,
+    type ConfigOptions,
+  } from '@forgerock/javascript-sdk';
+  import type { StepOptions } from '@forgerock/javascript-sdk/lib/auth/interfaces';
+  import { get } from 'svelte/store';
+
+  // Import store types
+  import type { JourneyStore, JourneyStoreValue } from '$journey/journey.store';
+  import type { OAuthStore, OAuthTokenStoreValue } from '$lib/oauth/oauth.store';
+  import type { UserStore, UserStoreValue } from '$lib/user/user.store';
+
+  import './main.css';
+
+  export interface Response {
+    journey?: JourneyStoreValue;
+    oauth?: OAuthTokenStoreValue;
+    user?: UserStoreValue;
+  }
+  export interface JourneyOptions {
+    config?: StepOptions;
+    oauth?: boolean; // defaults to true
+    user?: boolean; // defaults to true
+  }
+
+  let dialogComp: SvelteComponent;
+  let dialogEl: HTMLDialogElement;
+  let callMounted: (dialog: HTMLDialogElement, form: HTMLFormElement) => void;
+  let journeyStore: JourneyStore;
+  let oauthStore: OAuthStore;
+  let returnError: (response: Response) => void;
+  let returnResponse: (response: Response) => void;
+  let userStore: UserStore;
+
+  export const journey = {
```

Comment:
Again, this is huge and duplicated, but I just didn't have the time to figure out how to break this out without breakage.